subsys/bluetooth/host/direction.c

```diff
@@ -429,6 +459,139 @@ static int hci_df_set_conn_cte_tx_param(struct bt_conn *conn, uint8_t cte_types,
 }
 #endif /* CONFIG_BT_CTLR_DF_CONN_CTE_RSP */
 
+#if defined(CONFIG_BT_DF_CONNECTION_CTE_RX)
+static void prepare_conn_cte_rx_enable_cmd_params(struct net_buf *buf, struct bt_conn *conn,
+						  const struct bt_df_conn_cte_rx_param *params,
+						  bool enable)
+{
+	struct bt_hci_cp_le_set_conn_cte_rx_params *cp;
+	const uint8_t *ant_ids;
+
+	cp = net_buf_add(buf, sizeof(*cp));
+	(void)memset(cp, 0, sizeof(*cp));
+
+	cp->handle = sys_cpu_to_le16(conn->handle);
+	cp->sampling_enable = enable ? 1 : 0;
+
+	if (enable) {
+		uint8_t *dest_ant_ids;
+
+		if (params->cte_type & BT_DF_CTE_TYPE_AOA) {
+			cp->slot_durations = params->slot_durations;
+			cp->switch_pattern_len = params->num_ant_ids;
+			ant_ids = params->ant_ids;
+		} else {
+			/* Those values are put here due to constraints from HCI command
+			 * specification: Bluetooth Core Spec. 5.3 Vol 4,Part E, sec 7.8.85.
+			 * There is no right way to successfully send the command to enable CTE
+			 * receive for AoD mode (e.g. device equipped with single antenna).
+			 */
+			cp->slot_durations = BT_HCI_LE_ANTENNA_SWITCHING_SLOT_2US;
+			cp->switch_pattern_len = ARRAY_SIZE(df_dummy_switch_pattern);
+			ant_ids = &df_dummy_switch_pattern[0];
```

Comment:
Given that `ant_ids` is just a stack variable, and the value it's pointing to is copied to `dest_ant_ids`, `df_dummy_switch_pattern` could technically also be moved to a stack variable as it doesn't need to be static.

Reply:
With cost of 2 bytes of SRAM there is less instructions executed every time the function is called. Are you sure that it is worth it?

Reply:
> Are you sure that it is worth it?
Nope :D

As with most things, there's is a tradeoff between cycles and memory.
If this is something that is called often, then the static array is probably better. If this is rarely called, it may be worth doing the stack allocated array.